src/main/java/seedu/bookmark/logic/parser/GoalCommandParser.java

```diff
@@ -0,0 +1,47 @@
+package seedu.bookmark.logic.parser;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.bookmark.commons.core.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
+import static seedu.bookmark.logic.parser.CliSyntax.PREFIX_DEADLINE;
+import static seedu.bookmark.logic.parser.CliSyntax.PREFIX_PAGE;
+
+import seedu.bookmark.commons.core.Messages;
+import seedu.bookmark.commons.core.index.Index;
+import seedu.bookmark.logic.commands.GoalCommand;
+import seedu.bookmark.logic.parser.exceptions.ParseException;
+import seedu.bookmark.model.book.Goal;
+
+
+public class GoalCommandParser implements Parser<GoalCommand> {
+
+    @Override
+    public GoalCommand parse(String args) throws ParseException {
+        requireNonNull(args);
+        ArgumentMultimap argMultimap =
+                ArgumentTokenizer.tokenize(args, PREFIX_PAGE, PREFIX_DEADLINE);
+
+        Index index;
+
+        // Check index
+        try {
+            index = ParserUtil.parseIndex(argMultimap.getPreamble());
+        } catch (ParseException pe) {
+            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, GoalCommand.MESSAGE_USAGE), pe);
+        }
+
+        // Check page and date
+        if (!(argMultimap.getValue(PREFIX_PAGE).isPresent() && argMultimap.getValue(PREFIX_DEADLINE).isPresent())) {
+            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, GoalCommand.MESSAGE_USAGE));
+        // Check if page and deadline provided matches page and deadline regex.
+        } else if (!argMultimap.getValue(PREFIX_DEADLINE).get().matches(Goal.DEADLINE_REGEX)
```

Comment:
An additional benefit to Le Yang's suggested implementation is that currently it seems that users can create a Goal with wrongly formatted deadline or page which would lead to problems when calling parseDeadline() or getPage(), as a validity check is not done before creation.

Reply:
Thanks for spotting that! I added a static parseGoal() in ParserUtil that checks not only the format but validity of the date provided as well :)